yodeploy/tests/test_util.py

```diff
@@ -77,18 +78,18 @@ def test_permission_squash(self):
             'PATH': os.environ['PATH'],
             'PYTHONPATH': ':'.join(sys.path),
         }
-        subprocess.check_call((
-            'fakeroot',
-            'python',
-            '-c', (
-                'import yodeploy.util, os; '
-                'yodeploy.util.extract_tar("%s", "%s"); '
-                's = os.stat("%s"); '
-                'assert s.st_uid == 0; '
-                'assert s.st_gid == 0'
-            ) % (self.tmppath('test.tar.gz'), self.tmppath('extracted'),
-                 self.tmppath('extracted/bar'))
-        ), env=env)
+        test_script = (
+            'import yodeploy.util, os; '
+            'yodeploy.util.extract_tar("%s", "%s"); '
+            's = os.stat("%s"); '
+            'assert s.st_uid == 0; '
+            'assert s.st_gid == 0'
+        ) % (
+            self.tmppath('test.tar.gz'), self.tmppath('extracted'),
+            self.tmppath('extracted/bar'))
+
+        subprocess.check_call(
+            ('fakeroot', sys.executable, '-c', "'%s'" % test_script), env=env)
```

Comment:
This had me scratching my head, because it works as is for you and on envs. It failed for me though. It might be due to differences in fakeroot behavior between versions but I'm really not sure. This is what I'm seeing with the version I installed on my machine via homebrew:

```python
>>> import subprocess, sys
>>> foo = 'print("hi")'
'print("hi")'
>>> subprocess.check_call(('fakeroot', sys.executable, '-c', foo))
/usr/local/bin/fakeroot: eval: line 60: syntax error near unexpected token `('
/usr/local/bin/fakeroot: eval: line 60: `set --  -- /Users/thomas/.pyenv/versions/yodeploy-3.5.1/bin/python -c print("hi")'
```

The subprocess hangs indefinitely at this point, so ^C:

```python
^CTraceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/thomas/.pyenv/versions/3.5.1/lib/python3.5/subprocess.py", line 579, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/Users/thomas/.pyenv/versions/3.5.1/lib/python3.5/subprocess.py", line 562, in call
    return p.wait(timeout=timeout)
  File "/Users/thomas/.pyenv/versions/3.5.1/lib/python3.5/subprocess.py", line 1651, in wait
    (pid, sts) = self._try_wait(0)
  File "/Users/thomas/.pyenv/versions/3.5.1/lib/python3.5/subprocess.py", line 1601, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
>>> subprocess.check_call(('fakeroot', sys.executable, '-c', "'%s'" % foo))
hi
0
```

```bash
> fakeroot -v
fakeroot version 1.20.2
> brew info fakeroot
fakeroot: stable 1.20.2 (bottled)
Provide a fake root environment
https://tracker.debian.org/pkg/fakeroot
/usr/local/Cellar/fakeroot/1.20.2 (24 files, 195K) *
  Poured from bottle on 2016-09-29 at 15:04:58
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/fakeroot.rb
```

```bash
thomas@dougmorelikeugh:~$ fakeroot -v
fakeroot version 1.18.2
```